### PR TITLE
Ask before cancelling an ongoing upload

### PR DESCRIPTION
**Pull request by skjnldsv**
Requested for 15.

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

File: apps/files/js/file-upload.js [hunk 1]
```diff
@@ -1055,6 +1069,7 @@ OC.Uploader.prototype = _.extend({
 				 */
 				stop: function(e, data) {
 					self.log('stop', e, data);
+					self._uploading = false;
```
Comment (ChristophWurst) [inline apps/files/js/file-upload.js]: should it also be reset on `done` and/or `fail`?

Comment (skjnldsv) [reply apps/files/js/file-upload.js]: No, done is for individual upload, I want this only when everything is done. :)
But fail is breaking the full upload it seems, let me look!

Comment (skjnldsv) [reply apps/files/js/file-upload.js]: Failure also trigger done. I fixed the error handling :)

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by MorrisJobke** [comment]
files tags acceptance tests are unrelated

**Comment by MorrisJobke** [comment]
> Requested for 15.

Wasn't it requested for stable14? Mind to back port this?

**Comment by skjnldsv** [comment]
> > Requested for 15.
> 
> Wasn't it requested for stable14? Mind to back port this?

Checking w/ prof serv to make sure

**Pull request merged**